hoomd/md/pair/aniso.py

```diff
@@ -247,3 +250,248 @@ def type_shapes(self):
+        alj.shape["A"] = dict(vertices=cube_verts,
+                              faces=cube_faces,
+                              rounding_radii=1)
+
+    Warning:
+        Changing dimension in a simulation will invalidate this force and will
+        lead to error or unrealistic behavior.
+    """
+
+    # We don't define a _cpp_class_name since the dimension is a template
+    # parameter in C++, so use an instance level attribute instead that is
+    # created in _attach based on the dimension of the associated simulation.
+
+    def __init__(self, nlist, default_r_cut=None, mode='none'):
+        super().__init__(nlist, default_r_cut, mode)
+        params = TypeParameter(
+            'params', 'particle_types',
+            TypeParameterDict(epsilon=float,
+                              sigma_i=float,
+                              sigma_j=float,
+                              alpha=int,
+                              contact_ratio_i=0.15,
+                              contact_ratio_j=0.15,
+                              average_simplices=True,
+                              len_keys=2))
+
+        shape = TypeParameter(
+            'shape', 'particle_types',
+            TypeParameterDict(vertices=[(float, float, float)],
+                              faces=[[int]],
+                              rounding_radii=OnlyIf(
+                                  to_type_converter((float, float, float)),
+                                  preprocess=self._to_three_tuple),
+                              len_keys=1))
+
+        self._extend_typeparam((params, shape))
+
+    @staticmethod
+    def get_ordered_vertices(vertices, return_faces=True):
+        """Compute vertices and faces of a convex hull of given vertices.
+
+        Warning:
+            This method requires the ``coxeter`` package which is available on
+            PyPi and conda-forge by that name, and can be found on GitHub under
+            ``glotzerlab/coxeter``.
+
+        Args:
+            vertices (:math:`(N_v, 3)` numpy.ndarray of float): The vertices to
+                take the convex hull of and get ordered vertices and faces from.
+            return_faces (`bool`, optional): Whether to return faces as a list
+                of list of int which index into the returned vertices. Defaults
+                to ``True``. If ``False`` only vertices are returned and the
+                return type is not a tuple.
+
+        Returns:
+            tuple: A tuple containing:
+
+                * ``vertices`` (:math:`(N_v, 3)` `numpy.ndarray` of `float`) -
+                  The vertices of the convex hull.
+                * ``faces`` (`list` [`list` [`int`]]) - The indices into the
+                  vertices of vertices defining the faces.
+        """
+        try:
+            import coxeter
+        except ImportError as error:
+            raise RuntimeError(
+                "Method requires coxeter as a dependency.") from error
```

Comment:
Allow Python's `ImportError` to raise to the user's script. The message it provides may help the user understand why `coxeter` cannot be imported.

Reply:
from error will point to the original error while adding our explanation.

Reply:
I wasn't aware of this. There are many places in HOOMD we could use `from error` to help users and developers understand exceptions, such as the `AttributeError`s that we catch from C++ and re-raise as a different error.